unstructured/ingest/main.py

```diff
@@ -303,17 +343,52 @@ def main(
                 f"Preserving downloaded files but --download-dir is not specified,"
                 f" using {download_dir}",
             )
-    if s3_url:
-        doc_connector = S3Connector(
-            config=SimpleS3Config(
-                path=s3_url,
-                access_kwargs={"anon": s3_anonymous},
-                download_dir=download_dir,
-                output_dir=structured_output_dir,
-                re_download=re_download,
-                preserve_downloads=preserve_downloads,
-            ),
-        )
+    if remote_url:
+        protocol = urlparse(remote_url).scheme
+        if protocol in ("s3", "s3a"):
+            doc_connector = S3Connector(  # type: ignore
+                config=SimpleS3Config(
+                    path=s3_url,
+                    access_kwargs={"anon": s3_anonymous},
+                    download_dir=download_dir,
+                    output_dir=structured_output_dir,
+                    re_download=re_download,
+                    preserve_downloads=preserve_downloads,
+                ),
+            )
+        elif protocol in ("abfs", "az"):
+            doc_connector = AzureBlobStorageConnector(  # type: ignore
```

Comment:
This `AzureBlobStorageConnector` will, as it is simply an empty subclass of `FsspecConnector`, make instances of `FsspecIngestDoc`, rather than of `AzureBlobStorageIngestDoc`. As a result, the latter class is never used, and its `@requires_dependencies(["adlfs", "fsspec"], extras="azure")` is never called.
This last bit is not really a problem, as there is also a `requires_dependencies` decorator around `AzureBlobStorageConnector`.

Reply:
Hmm true, maybe we can parametrize that to provide the `...IngestDoc` class to the `FsspecConnector` so that it's used instead of the default `FsspecIngestDoc`, or maybe we can just skip it and remove both `S3IngestDoc` and `AzureBlobStorageIngestDoc`?

Reply:
I think both solutions are reasonable. The first one would be as simple as:
```python
# azure.py
@requires_dependencies(["adlfs", "fsspec"], extras="azure")
class AzureBlobStorageConnector(FsspecConnector):
    ingest_doc_class = AzureBlobStorageIngestDoc
```

```python
# ffspec.py
class FsspecConnector(BaseConnector):
    """Objects of this class support fetching document(s) from"""

    ingest_doc_class = FsspecIngestDoc

...

    def get_ingest_docs(self):
        return [
            self.ingest_doc_class(
                self.config,
                file,
            )
            for file in self._list_files()
        ]
```

```python
# s3.py
@requires_dependencies(["s3fs", "fsspec"], extras="s3")
class S3Connector(FsspecConnector):
    ingest_doc_class = S3IngestDoc
```

And the second one is as simple as just removing the ingest classes from existence. Perhaps the first approach allows for more flexibility in case there ever comes an `ffspec`-supported ingest connector that wants to do some additional processing? If that never happens though, then the second approach is simpler. 
I think it's best to defer to the maintainers here.

Reply:
So what I did is basically the same as you mentioned above but using `ingest_doc_cls` as an attribute 😆 But maybe makes more sense to just remove it and rely on `FsspecIngestDoc`?